### ISSUE TITLE
Steam Truck access

### ISSUE THREAD

**Issue by Myrhaug** [StrangeLoopGames/EcoIssues#8625]
We noticed an issue with my steam truck, a player could use my Steam Truck without having any form of access to my deeds. I tried to pick up and place again, he still got access.
I tried to change from inherited to whitelist too, but he didn't have access anywhere (I checked all my deeds in Real Estate).

Look at the attached images
![steam-truck-inherited](https://user-images.githubusercontent.com/40342749/42934228-f3f7f04e-8b46-11e8-9ea8-df653d07d898.png)
![steam-truck-inherited-2](https://user-images.githubusercontent.com/40342749/42934229-f41751fa-8b46-11e8-9a79-e479f130c550.png)
![steam-truck-whitelist](https://user-images.githubusercontent.com/40342749/42934230-f43b51cc-8b46-11e8-9392-ec098e3ecadf.png)

**Comment by MarjieAVolk** [comment]
Hmmm, I have been trying to get this to happen with no success.  The fact that you could actually pick it up and put it back down and still have this problem is pretty odd.

Is it still happening?  Would you be able to send me the save file, as well as the name of the friend who had access?

**Comment by Myrhaug** [comment]
Yup, he got access to all of my vehicles (no property). I cant see his name on any deeds in real estate desk.
Another problem we notices, my excavator got turned into his property with no pickup.

I will download game file later and publish a link to it abit later.

**Comment by MarjieAVolk** [comment]
So this one person always gets access to all of your vehicles, but not your land.  And no one else gets access to your vehicles?

I have one thing to check - if you search for the player who has access in the deed's whitelist, do they show up in the "Unauthorized" section?  Or are they missing completely?

**Comment by Myrhaug** [comment]
As you can see, Amphysbaenas is not autorized. I checked one time more in real estate desk, and he is on the unautorized tab. 

![20180724225425_1](https://user-images.githubusercontent.com/40342749/43165893-5ffd39b4-8f95-11e8-97c8-b2b1550445a3.jpg)
![20180724225441_1](https://user-images.githubusercontent.com/40342749/43165895-601afaee-8f95-11e8-8c64-a0f9dd54eb2f.jpg)

**Link to Save file + DB**
https://drive.google.com/open?id=1VrNqTl38nqIxwjG5R26_KAIvyysf1fq5

**Comment by MarjieAVolk** [comment]
Okay, looks like what's going on has to do with contracts.  Amphysbaenas has accepted this contract, and it's giving them access to your vehicles, presumably based on their position.  It looks like the build road contract will give the contractor access to any vehicles for as long as they are on the property specified.  It looks like this contract doesn't have any property specified though actually, so it is giving them access to any of your vehicles that are also not on any property.
![image](https://user-images.githubusercontent.com/23460210/43168751-41c8781a-8f53-11e8-9655-7ef9307498a2.png)

**Comment by Myrhaug** [comment]
That is wierd, then this issue will be related to #8626 ?
I could try to mark contract as finished.

**Comment by MarjieAVolk** [comment]
Yeah, for now I'd recommend that you try to cancel it, if you don't want this person to have access.  I will work on fixing this though, this isn't correct behavior.
